### DISCUSSION TITLE
Allow metrics to be credited to multiple sites/collectives

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#883] [Anything Else]
"Can I "stack" sites? Using GA, we can register a hit on multiple universal access ids:

ga('create', 'UA-XXX-AA', {'name': 'primary'});
ga('primary.send', 'pageview');

ga('create', 'UA-YYY-BB', {'name': 'networkOverall'});
ga('networkOverall.send', 'pageview');

ga('create', 'UA-ZZZ-CC', {'name': 'channelAnalytics'});
ga('channelAnalytics.send', 'pageview');
</script>

Is there a way I can do that plausible? Ie register a view on multiple site ids?

sometimes metrics need to credited to multiple sites/collectives.

In the example I provided:

First metrics was for platform
Second metrics was for a "network" or group of sites that work together
Third is for the specific site.

I could theoretically roll up the metrics personally, but then things like unique visitors doesn't quite work right as if A person went to site A and site B they would count as 2 uniques for the level above while in reality it was the same user looking at 2 properties."

(request via email)

**Update:** This use case is now documented at https://plausible.io/docs/plausible-script#can-i-send-stats-to-multiple-dashboards-at-the-same-time

**Comment by Vigasaurus** [comment]
I suppose the manual-mode that's been suggested a lot could help with something like this potentially, but in the interim someone could just load multiple scripts with differing `data-domain`s, no? 

It's not exactly a *good* solution since it wouldn't handle the plausible object super well; but if they just want the basic automatic pageview events it could work well enough.

**Reply by ukutaht** [reply]
That's right @Vigasaurus. Currently loading three scripts all with different `data-domain` should work OK. The problem and why I've said it's not supported is when you want to use custom events. If the user does not care about custom events then loading three separate scripts should work OK.

I don't like the way it's supported in the GA example shown. To deal with this specific use-case I would rather allow the data-domain to specify a list of site ids like this:

```html
<script data-domain="domain-one.com,domain-two.com,domain-three.com" src=".."></script>
```

The rest of the API would look exactly the same, nothing changes to the user but everything is inserted into three dashboards instead of 1.

This is preferable over loading three scripts because:
* Browser does not have to parse and execute 3 separate JS files
* Custom events will work as expected

And it's preferable over the GA way because the way you use it is exactly the same as with one dashboard. The GA example above looks like a hacky way to do something the API is not intended for.

**Reply by petethered** [reply]
Hi,

I'm the one that originally emailed @metmarkosaric with the question about stacking sites (in relation to the API to add sites).

In general, I wouldn't be using custom events in my use case as accessing simple metrics is the goal.  As per the email quoted, it's so that stats can be aggregated at differing levels.  For example, a website, a group of websites, and all products using the SAAS with users having different levels of access to metrics (site owner, network operator, admin)

> And it's preferable over the GA way because the way you use it is exactly the same as with one dashboard. The GA example above looks like a hacky way to do something the API is not intended for.

Are you referring to it being a hacky way of doing something the GA Api was not intended to do?  

https://developers.google.com/analytics/devguides/collection/analyticsjs/creating-trackers#working_with_multiple_trackers

The example in the quoted email is a direct implementation of a section of their documentation.

If you're referring to a hacky way of doing what the Plausible API was not intended to do, well... that's why I was asking of there was support for this particular use case.

**Reply by ukutaht** [reply]
@petethered The point I was trying to make is that the way of accomplishing this simple task is not very streamlined in GA. I understand it's the official way of doing it in GA but I question how much effort was put into making it as convenient and efficient as it can be from Google's side.

The fact that you must change how pageviews are sent seems inconvenient for the developer. Plus, it fires 3 requests for what could easily be accomplished by sending just 1 request to the server because the data is identical. This is why I say the API doesn't seem to be designed for this specific use-case. It could be a lot better and more efficient if it was designed (intended) to be used in exactly the way you're using it.

Looks like the way GA does it is optimized for flexibility in case you want to send different data to different properties. But with that they are sacrificing convenience for 90% of users who are probably sending identical data to all properties.

With Plausible we also have official workarounds for certain things in our documentation like tracking 404 pages. I have no problem calling our 404 tracking 'hacky', for example. I don't mean it in a disparaging way at all. I just use it to mean that it's a bit of a workaround and not exactly as polished as how we'd want to present a feature to the user.

If you don't use custom events, you can already send events to multiple dashboards in Plausible by including multiple scripts. It's also not an ideal solution for the reasons I mentioned in my original post but it works.

**Comment by metmarkosaric** [comment]
This use case is now documented at https://plausible.io/docs/plausible-script#can-i-send-stats-to-multiple-dashboards-at-the-same-time
